src/hooks/async/wallet.ts

```diff
@@ -41,15 +32,23 @@ export function useBalance(tokenAddress?: ContractAddress) {
     return undefined;
   }, [signer, chainId]);
 
+  // this is ugly but it works
+  const walletAddress = (walletSDK as any)?.connection?.signer?._address;
```

Comment:
is the problem that `walletSDK.getAddress()` can throw?

Reply:
the problem is that it's async

Reply:
In this flow I need it sync

Reply:
(and since *if there is a signer* the signer will always have the *address* property (can't have a signer that doesn't have an address) I get always get it sync this way)